pkg/chunked/storage_linux.go

```diff
@@ -151,9 +151,15 @@ func parsePullOptions(store storage.Store) pullOptions {
 
 func (c *chunkedDiffer) convertTarToZstdChunked(destDirectory string, payload *os.File) (int64, *seekableFile, digest.Digest, map[string]string, error) {
 	diff, err := archive.DecompressStream(payload)
-	if err != nil {
-		return 0, nil, "", nil, err
-	}
+        if err != nil {
+                return 0, nil, "", nil, err
+        }
+
+        defer func() {
+		if err := diff.Close(); err != nil {
+			return 0, nil, "", nil, err
+		}
```

Comment:
I’m not sure this can fail at all, but if it did, we should prefer the original error that caused this function to return, not a possibly-derived error that happens during a cleanup.

I think just a `defer diff.Close()` here would be good enough.